include/hw_constants_checks.h

```diff
@@ -17,12 +16,20 @@ namespace gbt{
      *  \param[in] phase Phase value to check.
      *  \return Returns `false` in case of success; `true` in case of error. The precise error is logged and written to the `error` RPC key.
      */
-    inline bool checkPhase(wisc::RPCMsg *response, uint8_t phase){
-        if (phase < gbt::PHASE_MIN)
-            EMIT_RPC_ERROR(response, stdsprintf("The phase parameter supplied (%hhu) is smaller than the minimal phase (%hhu).", phase, gbt::PHASE_MIN), true)
-        if (phase > gbt::PHASE_MAX)
-            EMIT_RPC_ERROR(response, stdsprintf("The phase parameter supplied (%hhu) is bigger than the maximal phase (%hhu).", phase, gbt::PHASE_MAX), true)
-        return false;
+    inline void checkPhase(uint8_t phase){
+        if (phase < PHASE_MIN)
+            std::stringstream errmsg;
+            errmsg.clear();
+            errmsg.str("");
+            errmsg << "The phase parameter supplied (" << phase << ") is smaller than the minimal phase (" << PHASE_MIN << ").";
+            throw std::range_error(errmsg)
```

Comment:
the type of the argument for the `std::exception` needs to be a `string` type, so convert the `stringstream` using the `.str()` method

Reply:
Also, if we get bored of declaring the `std::stringstream` and converting it to a string for every exception, we can create an helper templated function. It would take the elements which would be appended to the string stream as parameters and would return a string.
E.g.:
``` c++
throw std::range(utils::format_string("The phase parameter supplied (", std::setfill('0'), std::setw(2), phase, ") is smaller than the minimal phase (", PHASE_MIN, ").");
```

Reply:
I don't see much (any?) benefit to this, since the utils function basically has the same syntax as the `stringstream` construction, you save the definition of `stringstream` and clearing it out if you reuse it, but I wouldn't advocate for this change

Reply:
I wouldn't advocate for it either. ;-) It's only if we get bored of constructing the `std::stringstream` and then converting it to a `std::string`.